### ISSUE TITLE
[Tech Request]: more information about prepared statement in statement_info and processlist

### ISSUE THREAD

**Issue by aylei** [matrixorigin/matrixone#13368]
### Is there an existing issue for the same feature request?

- [X] I have checked the existing issues.

### Is your feature request related to a problem?

```Markdown
When inspecting prepared statement, `SHOW PROCESSLIST` and `system.statment_info` table does now record the original statement.


mysql> select statement_id, statement, duration / 1e6 as timeconsumed,
    -> cast(json_unquote(json_extract(stats, '$[2]')) / 1048576.00 as decimal(38,3)) as memorysize,
    -> response_at as collecttime,
    -> node_uuid as node,
    -> account,
    -> user,
    -> statement_type as type
    ->  from system.statement_info where response_at > '2023-12-11 04:00:00' and node_uuid='35626138-6138-6462-3364-326263313761' and response_at < now()and aggr_count = 0 order by duration desc limit 10;
+--------------------------------------+--------------------------+---------------+------------+----------------------------+--------------------------------------+---------+------+---------+
| statement_id                         | statement                | timeconsumed  | memorysize | collecttime                | node                                 | account | user | type    |
+--------------------------------------+--------------------------+---------------+------------+----------------------------+--------------------------------------+---------+------+---------+
| 590d099d-97df-11ee-9af2-06f37e31660c | execute __mo_stmt_id_284 | 235769.057704 |   1362.085 | 2023-12-11 04:43:58.028201 | 35626138-6138-6462-3364-326263313761 | sys     | dump | Execute |
| aec02b83-97e0-11ee-9af2-06f37e31660c | execute __mo_stmt_id_307 | 210904.297747 |    991.836 | 2023-12-11 04:53:06.440486 | 35626138-6138-6462-3364-326263313761 | sys     | dump | Execute |
| 0b5b8b42-97e0-11ee-9af2-06f37e31660c | execute __mo_stmt_id_296 | 188262.023611 |   1014.668 | 2023-12-11 04:48:09.670133 | 35626138-6138-6462-3364-326263313761 | sys     | dump | Execute |
| 59fb453a-97e2-11ee-9af2-06f37e31660c | execute __mo_stmt_id_362 | 153218.821871 |    909.695 | 2023-12-11 05:04:05.529455 | 35626138-6138-6462-3364-326263313761 | sys     | dump | Execute |
| df0f9918-97e1-11ee-9af2-06f37e31660c | execute __mo_stmt_id_347 | 147920.970678 |   1331.922 | 2023-12-11 05:00:34.005074 | 35626138-6138-6462-3364-326263313761 | sys     | dump | Execute |
| 53f5838d-97de-11ee-9af2-06f37e31660c | execute __mo_stmt_id_215 | 145734.669724 |    560.291 | 2023-12-11 04:35:09.954315 | 35626138-6138-6462-3364-326263313761 | sys     | dump | Execute |
| 40eab852-97dc-11ee-9af2-06f37e31660c | execute __mo_stmt_id_76  | 143587.342633 |    552.346 | 2023-12-11 04:20:16.866092 | 35626138-6138-6462-3364-326263313761 | sys     | dump | Execute |
| bf90b224-97dd-11ee-9af2-06f37e31660c | execute __mo_stmt_id_180 | 132520.161384 |    563.458 | 2023-12-11 04:30:47.776296 | 35626138-6138-6462-3364-326263313761 | sys     | dump | Execute |
| 716563a8-97dd-11ee-9af2-06f37e31660c | execute __mo_stmt_id_168 | 128088.317734 |    608.843 | 2023-12-11 04:28:32.198361 | 35626138-6138-6462-3364-326263313761 | sys     | dump | Execute |
| f53ae1c1-97db-11ee-9af2-06f37e31660c | execute __mo_stmt_id_62  | 123232.245128 |    874.532 | 2023-12-11 04:17:49.529508 | 35626138-6138-6462-3364-326263313761 | sys     | dump | Execute |
+--------------------------------------+--------------------------+---------------+------------+----------------------------+--------------------------------------+---------+------+---------+
```

```
mysql> select info from processlist() as a where node_id='35626138-6138-6462-3364-326263313761'\G
*************************** 1. row ***************************
info:
*************************** 2. row ***************************
info:
*************************** 3. row ***************************
info: execute __mo_stmt_id_502
*************************** 4. row ***************************
info:
*************************** 5. row ***************************
info:
*************************** 6. row ***************************
info: select
    						cron_task_id,
    						task_metadata_id,
    						task_metadata_executor,
    						task_metadata_context,
    						task_metadata_option,
    						cron_expr,
    						next_time,
    						trigger_times,
    						create_at,
    						update_at
						from mo_task.sys_cron_task
*************************** 7. row ***************************
info:
*************************** 8. row ***************************
info:
*************************** 9. row ***************************
info:
*************************** 10. row ***************************
info:
10 rows in set (0.06 sec)
```

### Describe the feature you'd like

`SHOW PROCESSLIST` and `system.statement_info` should record the original statement.

**Comment by LiSong0214** [comment]
两种方案：
1、每次写statement_info 时，把sql 拼到现在值后面，例如：execute __mo_stmt_id_284: select * from xxxxx
2、持久化 prepare 表（session_id, prepare_name, prepare_statement)。用户通过 join 查看
需要考虑两种方案的成本

**Comment by xzxiong** [comment]
related pr https://github.com/matrixorigin/matrixone/pull/13412

**Comment by aylei** [comment]
closed by #13412
